src/Entities/Player.cpp

```diff
@@ -36,6 +36,41 @@
 // 1000 = once per second
 #define PLAYER_LIST_TIME_MS std::chrono::milliseconds(1000)
 
+namespace
+{
+
+/** Returns the old Offline UUID generated before becoming vanilla compliant. */
+cUUID GetOldStyleOfflineUUID(const AString & a_PlayerName)
+{
+	// Use lowercase username
+	auto BaseUUID = cUUID::GenerateVersion3(StrToLower(a_PlayerName)).ToRaw();
+	// Clobber a full nibble around the variant bits
+	BaseUUID[8] = (BaseUUID[8] & 0x0f) | 0x80;
+
+	cUUID Ret;
+	Ret.FromRaw(BaseUUID);
+	return Ret;
+}
+
+
+
+
+
+/** Returns the folder for the player data based on the UUID given.
+This can be used both for online and offline UUIDs. */
+AString GetUUIDFolderName(const cUUID & a_Uuid)
+{
+	AString UUID = a_Uuid.ToShortString();
+
+	AString res("players/");
+	res.append(UUID, 0, 2);
+	res.push_back('/');
+	return res;
+}
```

Comment:
I think this should be a public API of the `cPlayer` class, even accessible from Lua, and it should already contain the `FILE_IO_PREFIX` in the returned path.

Reply:
I meant the plugins should have means to locate the file and folder where the player data is stored. Not the right PR for this, though.